retrofit/src/main/java/retrofit/converter/GsonConverter.java

```diff
@@ -45,69 +46,38 @@ public GsonConverter() {
    * decoding from JSON (when no charset is specified by a header) will use UTF-8.
    */
   public GsonConverter(Gson gson) {
-    this(gson, "UTF-8");
+    this(gson, Charset.forName("UTF-8"));
   }
 
   /**
    * Create an instance using the supplied {@link Gson} object for conversion. Encoding to JSON and
    * decoding from JSON (when no charset is specified by a header) will use the specified charset.
    */
-  public GsonConverter(Gson gson, String charset) {
+  public GsonConverter(Gson gson, Charset charset) {
     this.gson = gson;
     this.charset = charset;
+    this.mediaType = MediaType.parse("application/json; charset=" + charset.name());
```

Comment:
Something more granular would be nice. I'm building a string just to have it parsed and yanked apart.